### PR TITLE
fix: visibility issues for tabs panels and sidebars

### LINKED ISSUE #1352: Visibility of fields under tabs not working as intended
### Describe the bug
1) `visible` attribute on field definition has no effect when inside tabs.
2) `only_on` is not working for fields that are inside panels within tabs.

### Steps to Reproduce

1)

- Add a field under a tab, with `visible` block 
image
- Check show page:
image

2) 

- Add a field under a panel within a tab, with `only_on` attribute
image
- Check show page:
image

### Expected behavior & Actual behavior
Expected: visibility checks are respected
Actual: are not respected

### Notes
I have some guesses as what the causes could be. I am looking at the `visible_items` method in `tab.rb` (let me know if I am looking at the wrong place):
image
1) This method seems to only check for visibility for a view, not the fields' `visible` block
2) When there is no panel, `items` represents **fields**. However, when a panel is inside a tab `items` represents **panels**, so it does not do the view visibility checks. In this case the fields are found _one layer further_ `items.first.items_holder.items`

### System configuration
**Avo version**: 2.15.3

**Rails version**: 6.1.6.1

**Ruby version**: 3.0.2

#### Impact

 - [ ] High impact
 - [] Medium impact
 - [x] Low impact

#### Urgency

 - [ ] High urgency
 - [ ] Medium urgency
 - [x] low urgency

### PR DESCRIPTION
**Pull request by Paul-Bob**
# Description

Fixes #1352

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/docs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: spec/dummy/app/avo/resources/spouse_resource.rb [hunk 1]
```diff
@@ -9,4 +9,34 @@ class SpouseResource < Avo::BaseResource
 
   field :id, as: :id
   field :name, as: :text
+
+  # Uncomment this to test visibility inside tabs, panels and sidebar
+  # tabs do
+  #   field :hidden_field_inside_tabs, as: :text, visible: -> (resource:) { false }
+
+  #   tab "hidden testing visibility inside tabs" do
+  #     field :hidden_field_inside_tabs_inside_tab, as: :text, visible: -> (resource:) { false }
```
Comment (adrianthedev) [inline spec/dummy/app/avo/resources/spouse_resource.rb:18]: This is still visible on my end.

File: spec/dummy/app/avo/resources/spouse_resource.rb [hunk 2]
```diff
@@ -9,4 +9,34 @@ class SpouseResource < Avo::BaseResource
...
+
+  #     panel do
+  #       field :hidden_field_inside_tabs_inside_tab_inside_panel, as: :text, visible: -> (resource:) { false }
+  #     end
+  #   end
+  # end
+
+  # sidebar do 
+  #   field :hidden_field_inside_sidebar, as: :text, visible: -> (resource:) { false }
```
Comment (adrianthedev) [inline spec/dummy/app/avo/resources/spouse_resource.rb:27]: This is visible too.

**Review by adrianthedev — COMMENTED** [review]
GG on the visibility refactor for panel and tab 👌

I think we should have some tests for this feature.

File: spec/dummy/app/avo/resources/spouse_resource.rb (hunk 1 above)
Comment (Paul-Bob) [reply spec/dummy/app/avo/resources/spouse_resource.rb:18]: did you restart the local server before trying this one?

**Pull request merged**